ctl/string.h

```diff
@@ -6,29 +6,59 @@
 
 namespace ctl {
 
-struct string;
+class string;
 
 string
 strcat(const string_view, const string_view) noexcept __wur;
 
-struct string
+namespace __ {
+
+constexpr size_t string_size = 3 * sizeof(size_t);
+constexpr size_t sso_max = string_size - 1;
+constexpr size_t big_mask = ~(1ull << (8ull * sizeof(size_t) - 1ull));
```

Comment:
String needs to be 0x7fffffffffffffff characters long before it stops using sso?